### PR TITLE
Changed carousel JS to support multiple carousels on a page

### PR DESCRIPTION
**Pull request by sherakama**
@jbickar 

Steps to test:
1. Download and install stanford_slides and stanford_carousel
2. Create a bunch of slides
3. Place as many carousel blocks into regions as you can
4. View them in all their sliding glory.

### CODE REVIEW

**Comment by jbickar** [comment]
I'm going to hold on merging this PR until we can get it set up so that Jennifer can test it. I'm concerned of at least two things:
1. That multiple carousels on the page will introduce accessibility issues
2. That this specific implementation may introduce accessibility issues

**Comment by sherakama** [comment]
This is just step one for making this a much more fun module.

Right now you can have multiple of the same carousel! So not much use but it saves the page from breaking and paves the way for multiple carousels. 

CAROUSEL ALL THE... go home Shea.

**Comment by sherakama** [comment]
@jbickar, 

With Jennifer now no longer with our organization can we check this PR to ensure that the feature functions and come back to the multiple items on a page issue. There are limited use cases for having more than one on a page.

**Comment by jbickar** [comment]
Meh. Still not super-excited about this but I'm willing to be convinced.

We'll still need to test for accessibility.

**Comment by sherakama** [comment]
@jbickar ping on this pr:

Here is a use case to why we need the javascript to be self contained. Lets open up another issue for the accessibility stuff. https://sites.stanford.edu/jumpstart-cars/

**Comment by jbickar** [comment]
Something in this PR is causing the button titles to start with "Slide 0" instead of "Slide 1" when there is no caption. The three hardest things in computer science: naming things, and off-by-one errors.

File: js/stanford_carousel.js [hunk 1]
```diff
@@ -1,56 +1,76 @@
...
+
+    // put the ol.carousel-indicators inside the div.carousel-dots
+    $elem.find('div.view-footer div.carousel-dots').append('<ol class="carousel-indicators"></ol>');
+    var dots = [];
+
+    // If we ended up with an empty array, that means there are no h2 elements.
+    // Therefore, it's a carousel without captions.
+    $elem.find('.views-row').each(function(i, v) {
       // Call it "Slide 1", "Slide 2" ,etc.
       var header = "Slide " + i;
```
Comment (jbickar) [inline js/stanford_carousel.js]: i is 0 in this case, leading the header to start with "Slide 0". This needs to be i+1.

**Comment by jbickar** [comment]
carousels

2015-08-10-135350-screenshot

**Comment by jbickar** [comment]
Setting aside the inadvisability of having multiple carousels on a page, this change makes sense. I tested with multiple carousels on one page in VoiceOver on Safari and the nice thing is that keyboard focus stays within the given carousel when interacting with the controls of that carousel. 

Fix the indexing from 0 issue and we're GTG. Nice work.

**Comment by sherakama** [comment]
Great! Thanks for this review.

**Pull request merged**